src/crates-and-source-files.md

```diff
@@ -100,6 +83,48 @@ fn main() {
 }
 ```
 
+## Preludes and `no_std`
+
+All crates have a *prelude* that automatically inserts names from a specific
+module, the *prelude module*, into scope of each [module] and an [`extern
+crate]` into the crate root module. By default, the *standard prelude* is used.
+The linked crate is [`std`] and the prelude module is [`std::prelude::v1`].
```

Comment:
I had a question.  By default, aren't both `std` and `core` crates included in the extern prelude?  I also think `std` is in the prelude for `no_std`, but that might change soon: https://github.com/rust-lang/rust/pull/54658

Also, would it be helpful to describe the different preludes and scope layers?  It may not be super relevant, but it might help clarify precedence when there are name conflicts (additions to the standard prelude will not shadow extern crates).  IIUC, the layers are:
1. local variables
2. items in unnamed blocks
3. items in the current module
4. crate names in the **"extern prelude"**
5. standard library prelude
6. language prelude (built-in types)

(The reason I ask is that I was looking at updating `extern crate` for the edition, and I'm trying to wrap my head around it.)

Reply:
Yes, all that would be helpful. I didn't know what all the layers were, but if this is the correct list, it's definitely something that should be included.

For `core` being included with `std` in the extern prelude, I think that's all supposed to change with https://github.com/rust-lang/rust/issues/53128 but I'm not entirely sure.